espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/BackdoorUtils.java

```diff
@@ -1,53 +1,80 @@
-
-        return c;
-    }
 
-    public static Class<?> mapToBoxingClass(Class<?> c) {
-        if (c.equals(int.class)) return Integer.class;
-        if (c.equals(float.class)) return Float.class;
-        if (c.equals(double.class)) return Double.class;
-        if (c.equals(boolean.class)) return Boolean.class;
-        if (c.equals(long.class)) return Long.class;
-        if (c.equals(byte.class)) return Byte.class;
-        if (c.equals(short.class)) return Short.class;
-        if (c.equals(char.class)) return Character.class;
-
-        return c;
+    public static Object parseValue(Object o, Class<?> c) {
+        if (o == null) return null;
+        if (c.equals(boolean.class)) return Boolean.parseBoolean(o.toString());
+        if (c.equals(byte.class)) return Byte.parseByte(o.toString());
+        if (c.equals(short.class)) return Short.parseShort(o.toString());
+        if (c.equals(int.class)) return Integer.parseInt(o.toString());
+        if (c.equals(long.class)) return Long.parseLong(o.toString());
+        if (c.equals(float.class)) return Float.parseFloat(o.toString());
+        if (c.equals(double.class)) return Double.parseDouble(o.toString());
+        if (c.equals(char.class)) return o.toString().charAt(0);
+        return o;
     }
 
-    public static Object numericValue(Object o, Class<?> c) {
-        if (o == null) return null;
+    public static Class<?> parseType(final String className) {
+        switch (className) {
+            case "boolean":
+                return boolean.class;
+            case "byte":
+                return byte.class;
+            case "short":
+                return short.class;
+            case "int":
+                return int.class;
+            case "long":
+                return long.class;
+            case "float":
+                return float.class;
+            case "double":
+                return double.class;
+            case "char":
+                return char.class;
+            case "void":
+                return void.class;
+            default:
+                String fqn = className.contains(".") ? className : "java.lang.".concat(className);
+                try {
+                    return Class.forName(fqn);
+                } catch (ClassNotFoundException ex) {
+                    throw new IllegalArgumentException("Class not found: " + fqn);
+                }
+        }
+    }
 
-        if (c.equals(Integer.class)) return Integer.parseInt(o.toString());
-        if (c.equals(Float.class)) return Float.parseFloat(o.toString());
-        if (c.equals(Double.class)) return Double.parseDouble(o.toString());
-        if (c.equals(Long.class)) return Long.parseLong(o.toString());
-        if (c.equals(Byte.class)) return Byte.parseByte(o.toString());
-        if (c.equals(Short.class)) return Short.parseShort(o.toString());
 
-        return o;
+    public static List<BackdoorMethodArgs> parseArguments(List<BackdoorMethodArgs> arguments) {
```

Comment:
Is there any particular purpose in modifying the argument? Cannot we simply return the list instead?

Reply:
Not really, I want to get parsed `List<Object> values` and also `Class<?>[] types` types from the list of BackdoorMethodArgs.  Not sure what is the best way to do this, is it creating a new data class and return it Or creating a Map etc. Maybe you can suggest

Reply:
Anyways, changed it now